src/Middleware.php

```diff
@@ -110,23 +112,35 @@ public static function requestBuilder(callable $serializer)
      *
      * @return callable
      */
-    public static function signer(callable $credProvider, callable $signatureFunction)
+    public static function signer(callable $credProvider, callable $signatureFunction, $tokenProvider = null)
     {
-        return function (callable $handler) use ($signatureFunction, $credProvider) {
+        return function (callable $handler) use ($signatureFunction, $credProvider, $tokenProvider) {
             return function (
                 CommandInterface $command,
                 RequestInterface $request
-            ) use ($handler, $signatureFunction, $credProvider) {
+            ) use ($handler, $signatureFunction, $credProvider, $tokenProvider) {
                 $signer = $signatureFunction($command);
-                return $credProvider()->then(
-                    function (CredentialsInterface $creds)
-                    use ($handler, $command, $signer, $request) {
-                        return $handler(
-                            $command,
-                            $signer->signRequest($request, $creds)
-                        );
-                    }
-                );
+                if ($signer instanceof TokenAuthorization) {
```

Comment:
How does the return value of the signature function resolve to an instance of TokenAuthorization?